book-1-orientation/chapters/KILL_NICKELBACK_LINQ.md

```diff
@@ -0,0 +1,26 @@
+# Kill Nickelback
+
+In this exercise, you're going to use Dictionaries, LINQ, and Lists.
+
+## Setup
+
+```
+mkdir -p ~/workspace/csharp/exercises/nickelback && cd $_
+dotnet new console
+dotnet restore
+```
+
+## Instructions
+
+1. Define a List of Dictionaries, named `allSongs`, that contains 7 Dictionaries.
+1. Each dictionary in the List will have a key of type `string`, and a value of type `string`.
+1. The key will be the name of an artist.
+1. The value will be the song title by that artist. Make sure that some of the songs are from the band Nickelback. You can see a [list of their greatest hits](https://www.amazon.com/Best-Nickelback-1/dp/B00FFERTUK/) on Amazon.
+   ```cs
+   // Example
+   List<Dictionary<string, string>> allSongs = new List<Dictionary<string, string>>();
+   ```
+1. Once the `allSongs` List is populated with the 7 Dictionaries, filter that List using the LINQ method, .Where(), to find all of the Dictionaries where the artist is **_not_** Nickelback.
```

Comment:
why would we want a list of dictionaries?  wouldn't `Dictionairy<string,List<string>>` make more sense?  but then the exercise is really simple, so maybe this just doesn't make sense?  I may just be missing something, but I don't think this is right.

Reply:
Sorry, also, could you be meaning to use a `List<KeyValuePair<string,string>>`  Having a dictionary with a single value in it is not what you want though.

Reply:
Before this change the exercise already this sort of collection of dictionary things. Her change is basically only swapping out a `HashSet<>` for a `LIst<>`.

You make a good point about it being kinda awkward. Maybe we should create an issue to make a broader change?

Reply:
Ah, I see the closed pr now.  Yeah, I'd probably say this or the previous incarnation are super low value because the premise is not realistic.

Reply:
Realism wasn't the primary intent for these beginner exercises. It's just to practice using the data structures of the language, and strengthen their understanding how to write `if` statements and `foreach` iteration. We may find it awkward, but students don't have have enough understanding to see that.

All that said, completely open to alternative ideas for this exercise. It was thrown together as a fun exercise for them to work on several years ago now and can easily be replaced with something better.

Reply:
I've got this as the version that I use : 

# Kill Nickelback

In this exercise, you're going to use HashSets, and Lists.

## Instructions

1. Define a class called `Song` that has a string property `Artist` and a string property `Name`.  Both properties should be required (invariant) in order to construct an instance of `Song` 
1. Define a List, named `goodSongs`, that will hold instances of `Song`.
1. Define a HashSet, named `allSongs`, that contains 7 instances of `Song`.

    Make sure that some of the songs are from the band Nickelback. You can see a [list of their greatest hits](https://www.amazon.com/Best-Nickelback-1/dp/B00FFERTUK/) on Amazon.
1. Once the set is populated with 7 instances, iterate over the set of songs, and check if the band name is "Nickelback".
1. If the band is **not** Nickelback, then add it to the `goodSongs` list.
1. Use another `foreach` loop to print out all the good songs.

Reply:
I guess I also cover classes before collections, so that may not work for you all.